test/addons/async-hooks-promise/test.js

```diff
@@ -36,8 +36,12 @@ assert.strictEqual(
 
 hook1.disable();
 
-// Check that internal fields are no longer being set.
-assert.strictEqual(
-  binding.getPromiseField(Promise.resolve(1)),
-  0,
-  'Promise internal field used despite missing enabled AsyncHook');
+// Check that internal fields are no longer being set. This needs to be delayed
+// a bit because the `disable()` call only schedules disabling the hook in a
+// future microtask.
+setImmediate(() => {
```

Comment:
We can’t because we *need* the `pop_ids()` call to happen, so we need the `PromiseHook` working until at least the `kAfter` occurs.

This shouldn’t be a problem, it’s just emitting events for nobody to witness.